controllers/v1/pollsController.js

```diff
@@ -0,0 +1,565 @@
+            );
+
+            return resolve({
+                message: pollsList.message,
+                result: pollsList.data
+            });
+
+        } catch (error) {
+
+            return reject({
+                status: error.status || httpStatusCode.internal_server_error.status,
+                message: error.message || httpStatusCode.internal_server_error.message,
+                errorObject: error
+            });
+        }
+    })
+}
+
+
+     /**
+     * @api {get} /assessment/api/v1/polls/delete/:pollId Delete an poll
+     * @apiVersion 1.0.0
+     * @apiName Delete an poll
+     * @apiGroup Polls
+     * @apiHeader {String} X-authenticated-user-token Authenticity token
+     * @apiSampleRequest /assessment/api/v1/polls/delete/5b98fa069f664f7e1ae7498c
+     * @apiParamExample {json} Response:
+     * {
+     *  "status": 200,
+     *  "message": "Poll deleted successfully"
+     * }
+     * @apiUse successBody
+     * @apiUse errorBody
+     */
+     
+    /**
+    * Delete poll.
+    * @method
+    * @name delete
+    * @param {Object} req -request Data.
+    * @param {String} req.params._id - pollId.  
+    * @returns {String} - message
+    */
+
+   delete(req) {
+    return new Promise(async (resolve, reject) => {
+
+        try {
+
+            let result = await pollsHelper.delete(
+                req.params._id
+            );
+
+            return resolve({
+               message: result.message
+            });
+
+        } catch (error) {
+
+            return reject({
+                status: error.status || httpStatusCode.internal_server_error.status,
+                message: error.message || httpStatusCode.internal_server_error.message,
+                errorObject: error
+            });
+        }
+    })
+}
+
+   
+   /**
+     * @api {get} /assessment/api/v1/polls/getPollQuestions/:pollId Get the poll questions
+     * @apiVersion 1.0.0
+     * @apiName Get the poll questions
+     * @apiGroup Polls
+     * @apiHeader {String} X-authenticated-user-token Authenticity token
+     * @apiSampleRequest /assessment/api/v1/polls/getpollQuestions/5f2bcc04456a2a770c4a5f3b
+     * @apiParamExample {json} Response:
+     * {
+     *  "status": 200,
+     *  "message": "Poll questions fetched successfully",
+     *  "result": [{
+            "qid": "0c59eb10-e5cc-11ea-bd85-039b643a3785",
+            "question": "Which app do you use the most?",
+            "responseType": "radio",
+            "options": [{ "value": "","label":"samiksha"},
+                         { "value": "","label":"unnati"},
+                         { "value": "","label":"bodh"}] 
+     *     }]
+     * }
+     * @apiUse successBody
+     * @apiUse errorBody
+     */
+     
+    /**
+    * Get the poll questions
+    * @method
+    * @name getPollQuestions
+    * @param {Object} req -request Data.
+    * @param {String} req.params._id - pollId.  
+    * @returns {JSON} - poll questions and options
+    */
+
+   getPollQuestions(req) {
```

Comment:
@deepakr123 - In this function, we should return a shareable URL as well. Also, create a new API which given a poll id gives a shareable link of the same.

Reply:
resolved